apis/python/src/tiledbsoma/collection.py

```diff
@@ -107,10 +112,53 @@ def __init__(
         If None, the cache has not been loaded.
         """
 
+    @overload  # type: ignore[override]  # Bad typing in somacore.
+    def add_new_collection(
+        self,
+        key: str,
+        cls: None = None,
+        *,
+        uri: Optional[str] = ...,
+        platform_config: Optional[options.PlatformConfig] = ...,
+    ) -> "CollectionBase[AnyTileDBObject]":
+        ...
+
+    @overload
+    def add_new_collection(
+        self,
+        key: str,
+        cls: Type[_Coll],
+        *,
+        uri: Optional[str] = ...,
+        platform_config: Optional[options.PlatformConfig] = ...,
+    ) -> _Coll:
+        ...
```

Comment:
@thetorpedodog My only comment here is it's pretty hard to understand what's going on with the class hierarchy typing in `collection.py`. I think it'd be worth leaving a few breadcrumb comments to help the reader understand the goals & constraints that lead to the unusual constructs here (the three declarations of `add_new_collection`, `_real_class`, etc.). I don't doubt there's good reason for them, just need some signposts :smile_cat: 

I'm not familiar enough with TileDB Cloud to address the absolute/relative URI question I'm afraid...

Reply:
+1. Not familiar with the multiple overrides. Is this just for dev-time type-inference support for method return types?